fmn/delivery/backends/mail.py

```diff
@@ -0,0 +1,131 @@
+
+from fmn.lib import models
+from fmn.util import get_fas_email
+
+_log = logging.getLogger(__name__)
+
+
+CONFIRMATION_TEMPLATE = u"""
+{username} has requested that notifications be sent to this email address
+* To accept, visit this address:
+  {acceptance_url}
+* Or, to reject you can visit this address:
+  {rejection_url}
+Alternatively, you can ignore this.  This is an automated message, please
+email {support_email} if you have any concerns/issues/abuse.
+"""
+
+reason = u"""
+You received this message due to your preference settings at
+{base_url}{user}/email/{filter_id}
+"""
+
+
+class EmailBackend(BaseBackend):
+    __context_name__ = "email"
+
+    def __init__(self, *args, **kwargs):
+        super(EmailBackend, self).__init__(*args, **kwargs)
+        self.mailserver = self.config['fmn.email.mailserver']
+        if ':' in self.mailserver:
+            self.host, self.port = self.mailserver.split(':')
+        else:
+            self.host, self.port = self.mailserver, 25
+        self.port = int(self.port)
+        self.from_address = self.config['fmn.email.from_address']
+
+    @defer.inlineCallbacks
+    def deliver(self, formatted_message, recipient, raw_fedmsg):
+        """
+        Deliver a message to the recipient.
+
+        Args:
+            formatted_message (str): The formatted message that is ready for delivery
+                to the user. It has been formatted according to the user's preferences.
+            recipient (dict): The recipient of the message.
+            raw_fedmsg (dict): The original fedmsg that was used to produce the formatted
+                message.
+        """
+        try:
+            # TODO handle the mail server being down gracefully
+            yield smtp.sendmail(
+                self.host,
+                to_bytes(self.from_address),
+                [to_bytes(recipient['email address'])],
+                formatted_message,
+                port=self.port,
+            )
+            _log.info('Email successfully delivered to %s', recipient['email address'])
+        except smtp.SMTPBadRcpt as e:
+            _log.info('Failed to email %s: %s', recipient['email address'], str(e))
+            self.handle_bad_email_address(recipient)
+
+    def handle_confirmation(self, session, confirmation):
+        confirmation.set_status(session, 'valid')
+        acceptance_url = self.config['fmn.acceptance_url'].format(
+            secret=confirmation.secret)
+        rejection_url = self.config['fmn.rejection_url'].format(
+            secret=confirmation.secret)
+
+        template = self.config.get('fmn.mail_confirmation_template',
+                                   CONFIRMATION_TEMPLATE)
+        content = template.format(
+            acceptance_url=acceptance_url,
+            rejection_url=rejection_url,
+            support_email=self.config['fmn.support_email'],
+            username=confirmation.openid,
+        ).strip()
+        subject = u'Confirm notification email'
```

Comment:
Maybe drop "email" from the subject - the recipient probably knows it's an e-mail ☺

Reply:
I didn't write this, just moved it, but I think it means "Confirm the email address you entered for notifications", not "This is a confirmation notification in email format".